src/scripting_api/aform.js

```diff
@@ -564,7 +564,7 @@ class AForm {
           event.change.length +
           event.selStart -
           event.selEnd;
-        if (finalLen >= 8) {
+        if (finalLen > 8 || event.value[0] === "(") {
```

Comment:
I think the rhs of `||` is useless but definitely the lhs is a way better like this.

Reply:
The RHS is necessary because we validate every time the field changes. For final validation you're right that it's unnecessary, but for the validation-during-typing, without the RHS if someone starts typing `(888)` the validation will stop them - for some reason the first character seems to be ignored for validation-during-typing, but the second parenthesis will cause an error:
<img width="622" alt="Screen Shot 2021-11-24 at 3 34 02 PM" src="https://user-images.githubusercontent.com/479715/143325603-2f5b5e9b-ad81-468b-b3fa-fcf097d9af01.png">

This is because `(888)` only has 5 characters, so this code selects the second pattern (`999-9999`), and `(888)` does not match the beginning of the second pattern. So, in that case we want to select the first pattern to match against, which is what the RHS is doing.

Reply:
It makes sense, +1.